spec/Section 7 -- Response.md

```diff
@@ -60,33 +61,34 @@ response must not be empty. It must contain at least one error. The errors it
 contains should indicate why no data was able to be returned.
 
 If the `data` entry in the response is present (including if it is the value
-{null}), the `errors` entry in the response may contain any field errors that
-were raised during execution. If field errors were raised during execution, it
-should contain those errors.
+{null}), the `errors` entry in the response may contain a list of all _field
+error_ that were raised during execution. If any _field error_ was raised during
+execution, it should contain the `errors` entry.
 
 **Request errors**
 
-Request errors are raised before execution begins. This may occur due to a parse
-grammar or validation error in the requested document, an inability to determine
-which operation to execute, or invalid input values for variables.
+:: A _request error_ is an error raised during a _request_ but before execution
+begins. This may occur due to a parse grammar or validation error in the
+_Document_, an inability to determine which operation to execute, or invalid
+input values for variables.
```

Comment:
Note `CreateSourceEventStream()` can also raise a request error:

> 5. If groupedFieldSet does not have exactly one entry, raise a request error.
>
> -- https://spec.graphql.org/draft/#CreateSourceEventStream()

There's also various places where there are `Assert:` that could raise an error (though if you've done validation they should not occur, except [when we don't handle that well](https://github.com/graphql/graphql-spec/pull/860)).

The phrase "before execution begins" is quite a fuzzy definition it turns out; here's a comment where I try and pick it apart a bit: https://github.com/graphql/graphql-js/pull/3640#issuecomment-1153711229

My current opinion is that "execution begins"

- for queries: when you enter [ExecuteQuery()](https://spec.graphql.org/draft/#ExecuteQuery())
- for mutations: when you enter [ExecuteMutation()](https://spec.graphql.org/draft/#ExecuteMutation())
- for subscriptions: when you enter [MapSourceToResponseEvent()](https://spec.graphql.org/draft/#MapSourceToResponseEvent())

Reply:
Time boundary isn't the best way to describe this. Maybe prose about this being "typical" could be helpful, but really the difference is that field errors go alongside partial results where request errors do not

Reply:
That's kind of circular reasoning though; for example if an error is raised from the `subscribe()` resolver of a field (i.e. in step 10 of [CreateSourceEventStream()](https://spec.graphql.org/draft/#CreateSourceEventStream()) via the [ResolveFieldEventStream()](https://spec.graphql.org/draft/#ResolveFieldEventStream()) algorithm) then it's not clear from the spec whether that should be a request error (since it prevents setting up the subscription, so the request cannot continue) or a field error (since it originated from a field). And thus, it's not clear if it should resolve to `{data: null, errors: [...]}` or just `{errors: [...]}` - and thus not clear if it's a partial result or not.

In my opinion, the boundary is effectively that it's a request error if it happens before the first `ExecuteSelectionSet()`; the boundaries I defined above give this rule of thumb explicit positions in the algorithm that are passed once only for each request.